src/intents/schemas/lowLevel.ts

```diff
@@ -0,0 +1,117 @@
+
+const SignatureTypeEnumSchema = joi.string().valid(...signatureTypes);
+
+const TokenStandardSchema = joi.string().valid(...tokenStandards);
+
+const TokenJSONSchema = joi.object({
+  address: joi.ethereumAddress().required(),
+  standard: joi.number().required(),
+  idsMerkleRoot: joi.string().required(),
+  id: joi.string().required(),
+  disallowFlagged: joi.boolean().required()
+});
+
+const SegmentFunctionNameSchema = joi.string().valid(...segmentFunctionNames);
+
+const TokenStructSchema = joi.object({
+  addr: joi.ethereumAddress().required(),
+  standard: TokenStandardSchema.required(),
+  idsMerkleRoot: joi.string().required(),
+  id: joi.bigIntish().required(),
+  disallowFlagged: joi.boolean().required()
+});
+
+const IdsProofStructSchema = joi.object({
+  ids: joi.array().items(joi.string()).required(),
+  merkleProof_hashes: joi.array().items(joi.string()).required(),
+  merkleProof_flags: joi.array().items(joi.boolean()).required(),
+  statusProof_lastTransferTimes: joi.array().items(joi.string()).required(),
+  statusProof_timestamps: joi.array().items(joi.string()).required(),
+  statusProof_signatures: joi.array().items(joi.string()).required()
+});
+
+const FillStateParamsStructSchema = joi.object({
+  id: joi.string().required(),
+  startX96: joi.string().required(),
+  sign: joi.boolean().required()  
+});
+
+const SegmentStructSchema = joi.object({
+  data: joi.string().required(),
+  requiresUnsignedCall: joi.boolean().required()
+});
+
+const CallStructSchema = joi.object({
+  targetContract: joi.ethereumAddress().required(), // TODO: Check
+  data: joi.string().required()
+});
+
+
+const ContractCallParamSchema = joi.alternatives().try(
+  joi.bigIntish(),
+  joi.boolean(),
+  joi.string(),
+  SignatureTypeEnumSchema, 
+  TokenStructSchema,
+  IdsProofStructSchema,
+  FillStateParamsStructSchema,
+  SegmentStructSchema,
+  CallStructSchema,
+  // This is to avoid "Block-scoped variable 'ContractCallParamSchema' used before its declaration."
+  joi.array().items(() => ContractCallParamSchema)
+);
+
+const OracleJSONSchema = joi.object({
+  address: joi.ethereumAddress().required(),
+  params: joi.string().required()
+});
+
+const IdsProofJSONSchema = joi.object({
+  ids: joi.array().items(joi.string()).required(),
+  merkleProof_hashes: joi.array().items(joi.string()).required(),
+  merkleProof_flags: joi.array().items(joi.boolean()).required(),
+  statusProof_lastTransferTimes: joi.array().items(joi.string()).required(),
+  statusProof_timestamps: joi.array().items(joi.string()).required(),
+  statusProof_signatures: joi.array().items(joi.string()).required()
+});
+
+const FillStateParamsJSONSchema = joi.object({
+  id: joi.string().required(),
+  startX96: joi.string().required(),
+  sign: joi.boolean().required()  
+});
+
+const BitJSONSchema = joi.object({
+  index: joi.string().required(),
+  value: joi.string().required()
+});
+
+const SegmentParamValueSchema = joi.alternatives().try(
```

Comment:
Please, add some test with wrong parameters that fails at this level

Reply:
agree on this - and especially we should check that the error messages are meaningful. We had this issue with DSL schema where all Joi errors were from the top level and did not say which parameter the error happened at